### PR TITLE
Add custom metadata connector

### PR DESCRIPTION
**Pull request by mars-lan**
### 🤔 Why?

To support the attachment of custom metadata to tables.

### 🤓 What?

As titled.

### 🧪 Tested?

Manually ran the connector and verified the output.

### CODE REVIEW

File: metaphor/manual/metadata/extractor.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    CustomMetadataItem,
+    Dataset,
+    MetadataChangeEvent,
+)
+
+logger = get_logger(__name__)
+
+
+class CustomMetadataExtractor(BaseExtractor):
+    """Manual governance extractor"""
```
Comment (alyiwang) [inline metaphor/manual/metadata/extractor.py:18]: manual metadata?

File: metaphor/manual/metadata/extractor.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    @staticmethod
+    def from_config_file(config_file: str) -> "CustomMetadataExtractor":
+        return CustomMetadataExtractor(CustomMetadataConfig.from_yaml_file(config_file))
+
+    def __init__(self, config: CustomMetadataConfig) -> None:
+        super().__init__(config, "Custom metadata connector", None)
+        self._datasets = config.datasets
+
+    async def extract(self) -> List[MetadataChangeEvent]:
+        logger.info("Fetching governance from config")
```
Comment (alyiwang) [inline metaphor/manual/metadata/extractor.py:29]: metadata

File: metaphor/manual/metadata/extractor.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+        datasets = []
+        for dataset_metadata in self._datasets:
+            dataset = Dataset(logical_id=dataset_metadata.id.to_logical_id())
+            datasets.append(dataset)
+
+            dataset.custom_metadata = CustomMetadata(metadata=[])
+            for key, value in dataset_metadata.metadata.items():
+                dataset.custom_metadata.metadata.append(
+                    CustomMetadataItem(key=key, value=json.dumps(value))
```
Comment (alyiwang) [inline metaphor/manual/metadata/extractor.py:39]: Do we want to dedupe `key`?

Comment (mars-lan) [reply metaphor/manual/metadata/extractor.py:39]: No need since it came form a dict.

File: metaphor/manual/metadata/extractor.py (hunk 1 above)
Comment (mars-lan) [reply metaphor/manual/metadata/extractor.py:18]: I thought about the same, but Custom Metadata is aligned with the actual product feature. We might as well change the namespace to `metaphor.custom.*` for the other 2 connectors.

File: metaphor/manual/metadata/extractor.py (hunk 2 above)
Comment (mars-lan) [reply metaphor/manual/metadata/extractor.py]: Thanks for catching.

File: metaphor/manual/metadata/extractor.py (hunk 1 above)
Comment (alyiwang) [reply metaphor/manual/metadata/extractor.py:18]: At least governance -> metadata

Comment (mars-lan) [reply metaphor/manual/metadata/extractor.py]: Fixed.

**Review by alyiwang — APPROVED** [review]
LGTM

**Pull request merged**